src/services/member/service.ts

```diff
@@ -83,4 +102,60 @@ export class MemberService {
 
     return memberRepository.deleteOne(id);
   }
+
+  createEmailChangeRequest(member: Member, newEmail: string) {
+    const payload = { uuid: member.id, oldEmail: member.email, newEmail };
+    return jwt.sign(payload, EMAIL_CHANGE_JWT_SECRET, {
+      expiresIn: `${EMAIL_CHANGE_JWT_EXPIRATION_IN_MINUTES}m`,
+    });
+  }
+
+  /**
+   * Send an email to the member with a link to reset their password.
+   * The link targets to a frontend endpoint that will handle the password reset.
+   * @param email The email of destination
+   * @param token The JSON Web Token to reset the password.
+   * @param lang The language to use for the email.
+   * @param useDeprecated If true, the link will target the deprecated backend endpoint. Otherwise, it will target a frontend endpoint.
+   * @returns void
+   */
+  mailEmailChangeRequest(
+    newEmail: string,
+    token: string,
+    lang: string,
+    useDeprecated?: boolean,
+  ): void {
+    const translated = this.mailer.translate(lang);
+    const subject = translated(MAIL.CHANGE_EMAIL_TITLE);
+    const domain = useDeprecated ? ACCOUNT_HOST.url : PUBLIC_URL;
+    const path = useDeprecated ? '/email/change' : '/members/current/email/change';
+    const destination = new URL(path, domain);
+    destination.searchParams.set(SHORT_TOKEN_PARAM, token);
+    const link = destination.toString();
+
+    const html = `
+      ${this.mailer.buildText(translated(MAIL.CHANGE_EMAIL_TEXT))}
+      ${this.mailer.buildButton(link, translated(MAIL.CHANGE_EMAIL_BUTTON_TEXT))}
+      ${this.mailer.buildText(translated(MAIL.CHANGE_EMAIL_NOT_REQUESTED))}`;
+
+    const footer = this.mailer.buildFooter(lang);
+
+    // don't wait for mailer's response; log error and link if it fails.
+    this.mailer
+      .sendEmail(subject, newEmail, link, html, footer)
+      .catch((err) => this.log.warn(err, `mailer failed. link: ${link}`));
+  }
+
+  mailConfirmEmailChangeRequest(oldEmail: string, newEmail: string, lang: string) {
```

Comment:
```suggestion
  sendConfirmEmailChangeRequest(oldEmail: string, newEmail: string, lang: string) {
```

Reply:
Was it intentional to ignore this change ? I think having the method start with `send...` would be nice for consistency.